langserver/handler.go

```diff
@@ -397,7 +397,7 @@ func (h *LangHandler) Handle(ctx context.Context, conn jsonrpc2.JSONRPC2, req *j
 				// a user is viewing this path, hint to add it to the cache
 				// (unless we're primarily using binary package cache .a
 				// files).
-				if !h.Config.UseBinaryPkgCache || req.Method == "textDocument/didSave" {
+				if h.Config.DiagnosticsEnabled && (!h.Config.UseBinaryPkgCache || req.Method == "textDocument/didSave") {
```

Comment:
I'd suspect this matches the proper intent. We still want to run it if usebinarypkgcache is disabled
```go
!h.Config.UseBinaryPkgCache || (h.Config.DiagnosticsEnabled && req.Method == "textDocument/didSave")
```

Reply:
I have added the `Config.DiagnosticsEnabled` check in the `publishDiagnostics`. So there is no harm in doing the typecheck on save too, right?

Reply:
Anyway. I will change it to
```go
if !h.Config.UseBinaryPkgCache || (h.Config.DiagnosticsEnabled && req.Method == "textDocument/didSave") {
```